File: pc2/team/round3a-the-syn-strikes-back/challenge/java/encrypt.java
/*
Copyright 2022 Carnegie Mellon University.
Released under a MIT (SEI)-style license, please see LICENSE.md in the project 
root or contact permission@sei.cmu.edu for full terms.
*/

import java.util.*;
import java.io.*;
import java.lang.Math;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.nio.file.Files;
import java.io.IOException;
import java.nio.file.StandardOpenOption;

public class encrypt
{
    public static void main(String[] args) throws IOException
    {
        int c = 0;                                              //global value that increases with each iteration
        if (args.length != 1)
        {
            System.out.println("Please enter one argument");
            System.exit(0);
        }
        

        String s = args[0];
        if(s.charAt(0) == '-')
        {
            System.out.println("Cannot be a negative number");
            System.exit(0);
        }

        if (s.length() % 2 != 0)
        {
            System.out.println("string must be even length");
            System.exit(0);
        }

	if (s.length() < 2 || s.length() > 20)
        {
            System.out.println("string must be between 2 and 20 characters or less");
            System.exit(0);
	}

        for(int t = 0; t < s.length();t++) // check that string is only digits
        {
            if(Character.isDigit(s.charAt(t)))
            {
                continue;
            }
            else
            {
                System.out.println("String must contain numbers only");
                System.exit(0);
            }
        }

        String en = "";
        while ( s.length() > 0 )   
	    {
	        //int i = s.length();
    
     	    int a = Integer.parseInt(s.substring(0,1));
            int b = Integer.parseInt(s.substring(1,2));
           
	        int c1 = mapping.map(a);
            int c2 = mapping.map(b);

            c += a + b;
            char c3 = mapping.map(c);

            int d = c1 + c2;
            String temp = Integer.toString(d);
            
            temp += c3;

            en+=temp;
            
            s = s.substring(2); 
        }

    FileWriter fileWriter = new FileWriter("encryption.txt");
	fileWriter.write(en);
	fileWriter.close();
    }
}

File: pc2/team/round3a-the-syn-strikes-back/challenge/js/encrypt.js
/*
Copyright 2022 Carnegie Mellon University.
Released under a MIT (SEI)-style license, please see LICENSE.md in the project 
root or contact permission@sei.cmu.edu for full terms.
*/

"use strict";

const ALPHABET = Array.from("abcdefghijklmnopqrstuvwxyz");

Map.prototype.someName = function(a, b) {
    this.set(a, b)
}

window.onload = function() {
    let btn = document.getElementById("submitBtn");
    btn.addEventListener("click", takeInput);
    document.getElementById("textField").addEventListener("keydown", function(e) {
        if (!e) {
            e.preventDefault();
        }
        if (e.keyCode == 13) {
            takeInput();
        }
    }, false);
}

function takeInput() {
    let text = document.getElementById("textField").value;
    text = text.toLowerCase();

    if (text.length >= 13) {
        alert("string must not be longer than 12 characters");
        return;
    }
    for (let ch of text) {
        if (!isLetter(ch)) {
            alert("string must only be letters");
            return;
        }
    }

    let firstMap = createFirstMap(text)
    console.log(firstMap);
    let round1 = ""
    for (let ch of text) {
        round1 += firstMap.get(ch);
    }
    let round2 = "";
    for (let ch of round1) {
        round2 += ch + firstMap.get(ch);
    }

    let secondMap = createSecondMap(round2);
    console.log(secondMap);
    let round3 = "";
    for (let ch of round2) {
        round3 += secondMap.get(ch);
    }

    let final = "";
    for (let ch of round3) {
        final += ch + secondMap.get(ch);
    }
    console.log("Final " + final);
    alert(final);
}

function createFirstMap(text) {
    useless1(true, 17);
    let m = makeMap(-223);
    fillFirstMap(m, text);
    return m;
}

function fillFirstMap(m, text) {
    useless2(5);
    for (let [i, ch] of ALPHABET.entries()) {
        m.someName(ch, ALPHABET[(i + text.length) % 26]);
    }
}

function useless1(foo, bar) {
    if (bar >= 173548) {
        return (true && !!(true || false)) && (324673 & 12798125);
    } else {
        return foo == !!!!(true || false);
    }
}

function useless2(n) {
    switch(n) {
        case -245:
            console.log("ab");
            break;
        case -33:
            console.log("tt");
            break;
        case 2:
            console.log("asdbhjkg");
            break;
        case 5:
            if (navigator.userAgent.indexOf("Gecko") >=0) {
                useless3(true)
            } else {
                console.log("alsjhbfnljkds");
            }
    }
}

function useless3() {
    if (navigator.userAgent.indexOf("win64") >= 0) {
        console.log("jhsbnagdkj");
    } else {
        console.log("seadhfgk");
    }
}

function makeMap(n) {
    let a = 7518 * 3443;
    let b = 64867 / 56341;
    if (n > 200) {
        return !!(true||(false && false)) ? new Map() : null;
    } else {
        return !!(false && (true || (!false || true))) ? null : new Map();
    }
}

function createSecondMap(text) {
    let m = makeMap(352);
    fillSecond(m, text);
    return m;
}

function fillSecond(m, text) {
    for (let [i, ch] of ALPHABET.entries()) {
        if (crazyEvenCheck(i)) {
            let value = getValue(i, text)
            m.someName(ch, ALPHABET[value]);
        } else {
            m.someName(ch, ch);
        }
    }
}

function crazyEvenCheck(n) {
    let re = /(adjev)*fff(\d\d\d)?/
    if (re.test("adjevadjevadjevadjevfff197") && !(n % 2 !== 0)) {
        return true;
    }
    return false;
}

function getValue(i, text) {
    let shift = i - text.length;
    if (shift < 0) {
        shift = 26 - Math.abs(shift);
    }
    return shift;
}

function isLetter(str) {
    return str.length === 1 && str.match(/[a-z]/i);
}

